packages/commonwealth/server/controllers/server_groups_methods/get_group_members.ts

```diff
@@ -0,0 +1,57 @@
+import { WhereOptions } from 'sequelize';
+import { TypedPaginatedResult } from 'server/types';
+import { OrderByOptions } from '../../../../common-common/src/api/extApiTypes';
+import { CommunityInstance } from '../../models/community';
+import { MembershipAttributes } from '../../models/membership';
+import { formatPagination } from '../../util/queries';
+import { ServerGroupsController } from '../server_groups_controller';
+
+export type GetGroupMembersOptions = {
+  community: CommunityInstance;
+  groupId: number;
+  limit?: number;
+  page?: number;
+};
+
+export type GetGroupMembersResult = TypedPaginatedResult<MembershipAttributes>;
+
+export async function __getGroupMembers(
+  this: ServerGroupsController,
+  { community, groupId, limit, page }: GetGroupMembersOptions
+): Promise<GetGroupMembersResult> {
+  const group = await this.models.Group.findOne({
+    where: {
+      id: groupId,
+      chain_id: community.id,
+    },
+  });
```

Comment:
is this necessary? looks like `select * from membership where groupid = @groupId` is enough.

Reply:
Trying to understand the flows here, so we can properly place the membership entity in the model:
1. Whenever a group is created or modified, we are going to assign all community addresses to the group via membership
2. When a new address joins the community, we automatically link it to all groups in the community via membership
3. If a group gets deleted, no action required other than deleting group/membership

Is there a scenario where an address in a community can opt-out?

Reply:
I think that adding the chain ID to the query improves performance, haven't measured though.

That flow is accurate. I'm not aware of a way to remove an address from a community. From a UI standpoint, it looks like you can't do that.

Reply:
I'm questioning the need of group-address relations if you cannot opt-out. It will always be a copy of community-address.

Reply:
When a user initially joins a community, an Address is created. This makes a user a member of a community. This is the purpose of the `Addresses` table. The user cannot opt out of this.

However, the group-address relations are stored in the `Memberships` table. Since computing membership is expensive (requires 3rd party API calls), we persist the compute result in this table with a TTL. The "opting out" in this case happens implicitly when a membership becomes stale.

Reply:
Resolving so we can merge but cc @Rotorsoft re Ryan's reply.